src/pages/index.js

```diff
@@ -51,55 +73,154 @@ const popupCardForm = new PopupWithForm(editCardModal, handleCardFormSubmit);
+const popupProfileImageForm = new PopupWithProfilePicForm(
+  profileImageModal,
+  handleProfileImageFormSubmit
 );
-section.renderItems();
+popupProfileImageForm.setEventListeners();
+
+//PopupWithDelete Class
+const popupWithDeleteCard = new PopupWithDelete(
+  deleteCardModal,
+  handleDeleteCardFormSubmit
+);
+popupWithDeleteCard.setEventListeners();
 
 //UserInfo Class
-const userClass = new UserInfo({
+const userInfo = new UserInfo({
   nameContainer: profileName,
   descriptionContainer: profileDescription,
 });
 
+//Api Class
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "1755563a-df02-4ecd-aa91-5c5958bedadf",
+    "Content-Type": "application/json",
+  },
+});
+
+//Section Class
+const section = new Section({ renderer: renderCard }, cardListEl);
+
+api.getInitialCards().then((cardItems) => {
+  section.renderItems(cardItems);
+});
+
+api.getProfileInfo().then((res) => {
+  profilePic.src = res.avatar;
+
+  return userInfo.setUserInfo({
+    nameInput: res.name,
+    descriptionInput: res.about,
+  });
+});
+
+// initialCards.forEach((card) => {
+//   api.addCard(card);
+// });
+
 //
 //Functions
 function fillProfileInputs() {
-  const inputs = userClass.getUserInfo();
+  const inputs = userInfo.getUserInfo();
 
   profileNameInput.value = inputs.profileName;
   profileDescriptionInput.value = inputs.profileDescription;
 }
 
 function handleProfileFormSubmit(data) {
-  userClass.setUserInfo({
-    nameInput: data.title,
-    descriptionInput: data.description,
-  });
-
-  popupProfileForm.close();
+  api
+    .updateProfileInfo({
+      name: data.title,
+      about: data.description,
+    })
+    .then(() => {
+      userInfo.setUserInfo({
+        nameInput: data.title,
+        descriptionInput: data.description,
+      });
+      popupProfileForm.close();
+      setTimeout(() => {
+        editProfileButton.textContent = "Save";
+      }, 500);
+    });
 }
 
 function handleCardFormSubmit(data) {
   const name = data.title;
   const link = data.description;
-  renderCard({ name, link });
-  popupCardForm.close();
-  cardEditForm.reset();
-  addFormValidator.toggleButtonState();
+  api.addCard({ name: name, link: link }).then(() => {
+    renderCard({ name, link });
+    popupCardForm.close();
+    cardEditForm.reset();
+    addFormValidator.toggleButtonState();
+    setTimeout(() => {
+      addCardButton.textContent = "Create";
+    }, 500);
+  });
+  // renderCard({ name, link });
+  // popupCardForm.close();
+  // cardEditForm.reset();
+  // addFormValidator.toggleButtonState();
+}
+
+function handleDeleteCardFormSubmit(cardEl) {
+  api.deleteCard(cardEl._id).then(() => {
+    cardEl.remove();
+    setTimeout(() => {
+      deleteCardButton.textContent = "Yes";
+    }, 500);
+  });
+}
+
+function handleProfileImageFormSubmit(data) {
+  profilePic.src = profilePicModalInput.value;
+
+  api.updateProfilePic(data);
+
+  popupProfileImageForm.close();
```

Comment:
Needs correction: you should close the modal only after the successful api response